js-core/lesson3.5.md

```diff
@@ -0,0 +1,109 @@
+![](https://img.shields.io/badge/status-draft-darkred.svg) 
+
+
+# Pair Programming Day
+** What we will learn today?**
+- We will practice pair programming with a mentor
+---
+
+## How to work today?
+Today, we want to focus on our problem solving skills, how we think and tackle problems before even code, then crafting a **beautiful** solution using techniques like **TDD**
+- Think about the solution before writing any code (maybe even write pseudocode, or pencil a solution)
+- Break bigger tasks into smaller pieces, and solve them one step at a time.
+- Think about your function signature
+- Write Test -> Write Code Implementation -> Refactor (repeat)
+- Use the debugger
+- Good functions
+    - have good names and clear parameter names
+    - have single responsibility
+    - be defensive: don't trust your parameter
```

Comment:
Can we clarify this a bit?

- be defensive: your function might be called with a number when you expect a string, or your parameters might just be undefined. Don't trust your parameter!